### PR TITLE
Configuring beeper output with CLI

### PR DESCRIPTION
**Pull request by Pierre-A**
This was initiated in PR #1482 whose first motivation was more restrictive (and so is closed)

In order to easily configure the beeper output, I introduced the beeper_output_mode parameter in CLI whose value is one of four : N-PP , I-PP , N-OD , I-OD where PP/OD stands for (push-pull)NPN/PNP(opendrain) transistor and N/I for Normal/Inverted.
The default value comes from the target.h file through a #define BEEPER_OUTPUT_NPN for targets with NPN driver.

The buzzer output on targets not correctly identified (like Afromini, issue #1368) or with no driver on board (like CC3D) can then be accurately set.
For CC3D, it would be very easy to complete the lookup table for selecting the buzzer pin as two situations are yet encountered : directly soldered on STM pin38 (PA15) or using motor 6 output (PA2). Will be the object of a next coming PR.

Associated to the beeper_off_flags (see PR #1309), this gives a complete user-configurable buzzer function.

### CODE REVIEW

File: src/main/main.c [hunk 1]
```diff
@@ -279,21 +279,26 @@ void init(void)
...
-#ifdef BEEPER_INVERTED
-        .gpioMode = Mode_Out_PP,
-        .isInverted = true
-#else
-        .gpioMode = Mode_Out_OD,
-        .isInverted = false
-#endif
     };
-#ifdef NAZE
-    if (hardwareRevision >= NAZE32_REV5) {
```
Comment (ledvinap) [inline src/main/main.c]: Beeper won't work on older NAZE versions by default

**Comment by Pierre-A** [comment]
You're right, just for the first connecting time after flashing. 
But no more a problem once the `beeper_output_mode` parameter is set (among others).
Not sure this is worth coding that specific case.

**Comment by henn1001** [comment]
related https://github.com/borisbstyle/betaflight/commit/dfcc3eb02814a4bf0250ac3673aae98290ae244c
https://github.com/borisbstyle/betaflight/commit/d6c812f4fe1e7508e083afbe14d4411ef38ec906

**Comment by Pierre-A** [comment]
**@henn1001**, the present PR is not related to mentionned BB's ones as it allows configuring the output type (depending on the transistor driving the buzzer) and **not** the output line position. It could, but as BB has already treated the specific case of the CC3D board in Betaflight, I'm not sure it is necessary anymore (unless one think it should).

**Comment by henn1001** [comment]
Yeah i know. Just wanted to mention it because of the "next coming PR.". That should be fairly similar.

**Comment by Pierre-A** [comment]
Yes should be just an other way of coding but as I said, this is probably not necessary any more for betaflght and this has been delayed for cleanflight (issue #832).

**Comment by paddytfm** [comment]
hi everybody
beeper inverted command does not work on last naze32 2016 version
What can i do to make it work
Patrick

**Comment by Pierre-A** [comment]
**@paddytfm**, no you can't find this function in last release as it has not been merged in Cleanflight.

**Comment by paddytfm** [comment]
Ok Pierre-A
But any solution ?
Could I merge it ,if so what is the procedure
Patrick

**Comment by Pierre-A** [comment]
ah sorry, just forgot.. mail me, I can compile one for you after rebase on last cleanflight commit.

**Comment by paddytfm** [comment]
Thank's if you can do it for me
Patrick

**Comment by Pierre-A** [comment]
but mail me so I can send you the .hex file

**Comment by paddytfm** [comment]
Ok c'est fait a l'instant

**Comment by paddytfm** [comment]
Ok c’est fait y a quelques instant

 

De : Pierre-A [mailto:notifications@github.com] 
Envoyé : lundi 22 février 2016 11:41
À : cleanflight/cleanflight
Cc : paddytfm
Objet : Re: [cleanflight] Configuring beeper output with CLI (#1645)

 

but mail me so I can send you the .hex file

—
Reply to this email directly or view it on GitHub <https://github.com/cleanflight/cleanflight/pull/1645#issuecomment-187117886> .  <https://github.com/notifications/beacon/AQlp-l-PEN5vLEM2fslje794wrYxq2M3ks5pmt0tgaJpZM4G9Zxo.gif> 



---
L'absence de virus dans ce courrier électronique a été vérifiée par le logiciel antivirus Avast.
https://www.avast.com/antivirus

**Comment by paddytfm** [comment]
Bonjour Pierre
Suite a communication sur forum github, j’ai donc le fameux problème du transistor sur le beeper qui positionne celui-ci a ON au repos.
Vu que c’est un CMS et que ça bouste le buzzer ça me fait ??? de l’enlever ou dans rajouter un du coup une petite modif software serait la bienvenue.
Merci pour l’action.
A titre indicatif Juste une autre question comment on compile les src etc… cleanflight
Cordialement
Patrick

**Comment by paddytfm** [comment]
I just want to thanks’ Pierre for is action getting my Buzzer working fine.
Initially plug the buzzer was buzzing all the time because it is connected to my naze through a transistor reversing the default OFF state to ON state. Pierre did a bit of code to make it work properly. So the CLI command set beeper_output_mode = **** did the good job
Hope this command will be maintained in the new release of cleanflight and betaflight as well

**Comment by Pierre-A** [comment]
Great ! enjoy your flights.

**Comment by hydra** [comment]
related to #884 which should come first.

**Comment by airkaos** [comment]
Hello guys, am using betaflight on an afromini, but would love to go back to cleanflight, how could i get the .hex file with the beeper working for my afromini??

**Comment by RBirdie001** [comment]
Hi!
Also I'm waiting whether it will be merged. It's a shame throw the work whis was already done.
Roman

**Pull request closed without merging**